src/PublisherNodeConfiguration.cs

```diff
@@ -148,63 +145,81 @@ public static void Deinit()
                     try
                     {
                         await PublisherNodeConfigurationFileSemaphore.WaitAsync();
-                        _configurationFileEntries = JsonConvert.DeserializeObject<List<PublisherConfigurationFileEntry>>(File.ReadAllText(_publisherNodeConfigurationFilename));
+                        _configurationFileEntries = JsonConvert.DeserializeObject<List<PublisherConfigurationFileEntryLegacy>>(File.ReadAllText(PublisherNodeConfigurationFilename));
                     }
                     finally
                     {
                         PublisherNodeConfigurationFileSemaphore.Release();
                     }
-                    Trace($"Loaded {_configurationFileEntries.Count} config file entry/entries.");
 
-                    foreach (var publisherConfigFileEntry in _configurationFileEntries)
+                    if (_configurationFileEntries != null)
                     {
-                        if (publisherConfigFileEntry.NodeId == null)
+                        Logger.Information($"Loaded {_configurationFileEntries.Count} config file entry/entries.");
+                        foreach (var publisherConfigFileEntryLegacy in _configurationFileEntries)
                         {
-                            // new node configuration syntax.
-                            foreach (var opcNode in publisherConfigFileEntry.OpcNodes)
+                            if (publisherConfigFileEntryLegacy.NodeId == null)
                             {
-                                ExpandedNodeId expandedNodeId = ExpandedNodeId.Parse(opcNode.ExpandedNodeId);
-                                _nodePublishingConfiguration.Add(new NodePublishingConfiguration(expandedNodeId, publisherConfigFileEntry.EndpointUri, publisherConfigFileEntry.UseSecurity, opcNode.OpcSamplingInterval ?? OpcSamplingInterval, opcNode.OpcPublishingInterval ?? OpcPublishingInterval));
+                                // new node configuration syntax.
+                                foreach (var opcNode in publisherConfigFileEntryLegacy.OpcNodes)
+                                {
+                                    if (opcNode.ExpandedNodeId != null)
+                                    {
+                                        ExpandedNodeId expandedNodeId = ExpandedNodeId.Parse(opcNode.ExpandedNodeId);
+                                        _nodePublishingConfiguration.Add(new NodePublishingConfiguration(expandedNodeId, opcNode.ExpandedNodeId, publisherConfigFileEntryLegacy.EndpointUrl, publisherConfigFileEntryLegacy.UseSecurity, opcNode.OpcSamplingInterval ?? OpcSamplingInterval, opcNode.OpcPublishingInterval ?? OpcPublishingInterval));
+                                    }
+                                    else
+                                    {
+                                        // check Id string to check which format we have
```

Comment:
Similar code in a couple of places, might want to create helper or extension library.  Plus UrlDecode for non-json chars.

Reply:
will check this one later.